mkosi/config.py

```diff
@@ -1160,6 +1161,48 @@ def config_parse_minimum_version(value: Optional[str], old: Optional[GenericVers
     return max(old, new)
 
 
+def config_path_run(script: Path) -> str:
+    if not os.access(script, os.X_OK):
+        die(f"{script} is not executable")
+
+    return run([script.absolute()], stdout=subprocess.PIPE, env=os.environ).stdout
+
+
+def config_path_read(file: Path) -> str:
+    return file.read_text()
+
+
+def config_path_run_or_read(file: Path) -> str:
+    "Run the specified file and capture its output if it's executable, else read file contents"
+    if os.access(file, os.X_OK):
+        return config_path_run(file)
+
+    content = file.read_text()
+
+    if content.startswith("#!/"):
+        die(f"{file} starts with a shebang ({content.splitlines()[0]})",
+            hint="This file should be executable")
+
+    return content
+
+
+def config_make_path_handler(handler: ConfigPathCallback,
+                             strip_newlines: bool = True,
+                             allow_multiline: bool = True) -> ConfigPathCallback:
```

Comment:
Let's add these options when we actually need them. Since the values are always the same at the moment let's just drop them for now.